backend/src/main/java/ca/bc/gov/hlth/accounttransfer/controller/AccountsController.java

```diff
@@ -2,93 +2,94 @@
+
+	public AccountsController() {
+
+	}
 	@Autowired
-	private AccountsService accountsService;
+	private ClientsLookup clientsLookup;
+
+	@Autowired
+	private LdapService ldapService;
+
+	@Autowired
+	private KeycloakUserManagementService keycloakUserManagementService;
+
 	
 	/**
 	 * Sample post endpoint for the application 
 	 *  
-	 * @param the accountRequest
-	 * @return The sample response
+	 * @param accountTransferRequest account transfer request
+	 * @return The result of the operation
 	 */
-	@PostMapping("/get-account")
-	public ResponseEntity<AccountResponse> getAccount(@Valid @RequestBody AccountRequest accountRequest) {
-		logger.debug("getAccount request: {} ", accountRequest.getName());
-		
-		String accountId = accountsService.getAccountId(accountRequest.getName());				
-		AccountResponse accountResponse = new AccountResponse();
-		accountResponse.setId(accountId);
-		accountResponse.setMessage("Successful");
-		accountResponse.setStatus(StatusEnum.SUCCESS);
-		
-		ResponseEntity<AccountResponse> response = ResponseEntity.ok(accountResponse);
-		logger.debug("getAccount response: {} ", accountResponse.getId());
-		
-		return response;	
-	}
+	@PostMapping("/accountTransfer")
+	public ResponseEntity<AccountTransferResponse> transferAccount(@Valid @RequestBody AccountTransferRequest accountTransferRequest) {
 
-	/**
-	 * Sample get endpoint for the application 
-	 *  
-	 * @param the name of the account
-	 * @return The sample response
-	 */
-	@GetMapping("/{name}")
-	public ResponseEntity<AccountResponse> getAccount(@PathVariable("name") String name) {
-		logger.debug("getAccount name param: {} ", name);
-		
-		String accountId = accountsService.getAccountId(name);
-		
-		AccountResponse accountResponse = buildSampleResponse(name, accountId);
-		
-		ResponseEntity<AccountResponse> response = ResponseEntity.ok(accountResponse);
-		logger.debug("getAccount response: {} ", accountResponse.getId());
-		
-		return response;	
-	}
+		Jwt authToken = (Jwt) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+		String userId = authToken.getClaimAsString(subClaim);
+
+		logger.debug("Attempting account transfer for User {} on Application {}", accountTransferRequest.getUsername(), accountTransferRequest.getApplication());
+
+		LdapRequest ldapRequest = new LdapRequest(accountTransferRequest.getUsername(), accountTransferRequest.getPassword());
+		LdapResponse ldapResponse = ldapService.checkLdapAccount(ldapRequest);
+
+		// TODO if ldapResponse.username == '' -> error username doesn't exist
+		// TODO if ldapResponse.authenticated == false -> error user is locked
+		// TODO if ldapResponse.locked == true -> error account is locked
+		// TODO if ldapResponse.mspDirectRole = '' -> error user doesn't actually have the role
+		// TODO if ldapResponse.mspDirectRole is not supported in the new MSP Direct -> role is invalid
+
+		logger.debug("User has role {}", ldapResponse.getMspDirectRole().toUpperCase());
```

Comment:
move this to inside the the block of code that uses it
Once error handling is in place NPE checks should already be complete since a null role returned is an error case